src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,331 @@
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted);
+                Assert.True(client.IsMutuallyAuthenticated);
+                Assert.False(client.IsServer);
+                Assert.True(client.IsSigned);
+                Assert.False(client.LeaveInnerStreamOpen);
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated);
+                IdentityValidator.AssertHasName(serverIdentity, target);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthWithoutRealm_Success()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new UnixGssFakeNegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated);
+
+                Task[] auth = new Task[2];
+                NetworkCredential credential = new NetworkCredential(TestConfiguration.KerberosUser, TestConfiguration.Password);
+                auth[0] = client.AuthenticateAsClientAsync(credential, TestConfiguration.HostTarget);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated);
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted);
+                Assert.True(client.IsMutuallyAuthenticated);
+                Assert.False(client.IsServer);
+                Assert.True(client.IsSigned);
+                Assert.False(client.LeaveInnerStreamOpen);
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated);
+                IdentityValidator.AssertHasName(serverIdentity, TestConfiguration.HostTarget);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux | PlatformID.OSX)]
+        public void NegotiateStream_StreamToStream_KerberosAuthDefaultCredentials_Success()
```

Comment:
Let us add a DefaultCreds_failure test case where TGT does not exist before calling AuthenticateAsClientAsync